xcdat/spatial_avg.py

```diff
@@ -338,78 +339,136 @@ def _get_weights(
+        domain_bounds : xr.DataArray
+            The array of bounds for the latitude domain.
+        region_bounds : np.array
+            The array of bounds for latitude regional selection.
+
+        Returns
+        -------
+        xr.DataArray
+            The longitude axes weights.
         """
-        Ensure the domain bounds are within 0 to 360.
+        p_meridian_index: Optional[int] = None
+
+        if region_bounds is not None:
+            domain_bounds = self._swap_lon_axes(domain_bounds, to=360)
+            region_bounds = self._swap_lon_axes(region_bounds, to=360)
+
+            is_region_circular = region_bounds[1] - region_bounds[0] == 0
+            if is_region_circular:
+                region_bounds = np.array([0.0, 360.0])
+
+            (
+                domain_bounds,
+                p_meridian_index,
+            ) = self._align_longitude_to_360_axis(domain_bounds)
+            domain_bounds = self._scale_domain_to_region(domain_bounds, region_bounds)
+
+        weights = self._calculate_weights(domain_bounds)
+        if p_meridian_index is not None:
+            weights[p_meridian_index] = weights[p_meridian_index] + weights[-1]
+            weights = weights[0:-1]
+
+        return weights
+
+    def _get_latitude_weights(
+        self, domain_bounds: xr.DataArray, region_bounds: xr.DataArray
+    ) -> xr.DataArray:
+        """Gets weights for the latitude axes.
 
-        This method is designed to deal with the special case that
-        a grid cell encompasses the prime meridian (e.g., [359, 1]).
-        In this case, calculating longitudinal weights is complicated
+        This method scales the domain to a region (if selected). It scales the
+        area between two lines of latitude scales as the difference of the sine
+        of latitude bounds.
+
+        Parameters
+        ----------
+        domain_bounds : xr.DataArray
+            The array of bounds for the latitude domain.
+        region_bounds : np.array
+            The array of bounds for latitude regional selection.
+
+        Returns
+        -------
+        xr.DataArray
+            The latitude axes weights.
+        """
+        if region_bounds is not None:
+            domain_bounds = self._scale_domain_to_region(domain_bounds, region_bounds)
+
+        domain_bounds = np.sin(np.radians(domain_bounds))
+        weights = self._calculate_weights(domain_bounds)
+
+        return weights
+
+    def _calculate_weights(self, domain_bounds: xr.DataArray):
+        """Calculate weights for the domain.
+
+        This method takes the absolute difference between the upper and lower
+        bound values to calculate weights.
+
+        Parameters
+        ----------
+        domain_bounds : xr.DataArray
+            The array of bounds for a domain.
+
+        Returns
+        -------
+        xr.DataArray
+            The weights for an axes.
+        """
+        return np.abs(domain_bounds[:, 1] - domain_bounds[:, 0])
```

Comment:
This formula is abstracted into a separate method for reusability in `_get_longitude_weights()` and `get_latitude_weights()`.